api/clients/v2/verification/block_number_provider.go

```diff
@@ -0,0 +1,136 @@
+package verification
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/common"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+)
+
+// BlockNumberProvider is a utility for interacting with the ethereum block number
+type BlockNumberProvider struct {
+	logger    logging.Logger
+	ethClient common.EthClient
+	// duration of interval when periodically polling the block number
+	pollIntervalDuration time.Duration
```

Comment:
add comment that this can be negative. btw are we sure we need this behavior of alowing negative numbers here?

Reply:
Fair point - I've added checks so that duration <= simply causes errors [9db13f7f](https://github.com/Layr-Labs/eigenda/pull/1368/commits/9db13f7ff69160b739241e6b8171a5db07cb7f6d)